Lib/pathlib/types.py

```diff
@@ -14,10 +14,15 @@
 from glob import _PathGlobber
 from pathlib._os import magic_open, ensure_distinct_paths, ensure_different_files, copyfileobj
 from pathlib import PurePath, Path
-from typing import Optional, Protocol, runtime_checkable
+from typing import (
+    Any, BinaryIO, Callable, Generator, Iterator, Literal, Optional, Protocol, Self, Sequence, TypeVar,
+    runtime_checkable,
+)
 
+_WP = TypeVar("_WP", bound="_WritablePath")
 
-def _explode_path(path, split):
+
+def _explode_path(path: str, split: Callable[[str], tuple[str, str]]) -> tuple[str, list[str]]:
```

Comment:
Does it fit on 80 chars?

Reply:
(If you want to make `_explode_path()` accept a `parser: _PathParser` argument instead, that's fine, please go ahead)

Reply:
I changed the signature to accept a parser